### PR TITLE
[java][grid]: Set test name to video file name in dynamic grid

### PR DESCRIPTION
**Pull request by VietND96**
## **User description**
**Thanks for contributing to Selenium!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.

### Description
[java][grid]: Set test name to video file name in dynamic grid

### Motivation and Context
This is based on the support of [Metadata in tests](https://www.selenium.dev/documentation/grid/getting_started/#metadata-in-tests). When running test with dynamic grid, by adding metadata to your tests, the video file name will extract the value of capability `se:name` and then set to output the video file name.

For example, in Python binding:

```python
from selenium.webdriver.chrome.options import Options as ChromeOptions
from selenium import webdriver

options = ChromeOptions()
options.set_capability('se:recordVideo', True)
options.set_capability('se:screenResolution', '1920x1080')
options.set_capability('se:name', 'test_visit_basic_auth_secured_page (ChromeTests)')
driver = webdriver.Remote(options=options, command_executor="http://localhost:4444")
driver.get("https://selenium.dev")
driver.quit()
```

After test executed, under `/assets` you can see the video file name under `/<sessionId>/test_visit_basic_auth_secured_page_ChromeTests.mp4`

File name will be trimmed to 255 characters to avoid long file names. Moreover, `space` character will be replaced by `_` and only characters alphabets, numbers, `-` (hyphen), `_` (underscore) are retained in the file name.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

___

## **Type**
enhancement

___

## **Description**
- Refactored the handling of environment variables in `DockerSessionFactory` for better clarity and separation of concerns.
- Introduced functionality to set the video file name based on the `se:name` capability, enhancing traceability of test artifacts.
- Implemented sanitization and truncation for video file names to ensure compatibility and prevent excessively long file names.
- Updated the JDK version in the IDE configuration to JDK 17, aligning with updated language features and support.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>DockerSessionFactory.java</strong><dd><code>Refactor Env Vars Handling and Set Video File Name from Test Name</code></dd></summary>
<hr>
      
java/src/org/openqa/selenium/grid/node/docker/DockerSessionFactory.java

<li>Refactored environment variable setting into separate methods for <br>clarity.<br> <li> Added capability to set video file name based on the <code>se:name</code> <br>capability in session request.<br> <li> Video file name sanitization and truncation logic added.<br>

</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/13907/files#diff-a05d31304c9c75c40d9d11ef388a2c669f1f8c35e943ed8dcbc32e1da7694c5e">+41/-16</a>&nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Configuration changes
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>misc.xml</strong><dd><code>Update Project JDK Version in IDE Configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
.idea/misc.xml

- Updated project JDK version from 11 to 17 in IDE configuration.

</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/13907/files#diff-7ea1cd7bd5d16299a0abd33128e8d0a8bc04146f6b00c07bcd3cc0aad813c229">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by diemol** [comment]
I cannot update the fork with the latest from `trunk`.

**Comment by VietND96** [comment]
@diemol, I synced this, can you review this first?

**Review by diemol — APPROVED** [review]
Thank you, @VietND96!

File: .idea/misc.xml [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <component name="JavaScriptSettings">
     <option name="languageLevel" value="ES6" />
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_11" default="false" project-jdk-name="11" project-jdk-type="JavaSDK">
```
Comment (diemol) [inline .idea/misc.xml]: Wait, I forgot to mention. Can you please revert the changes to this file?

Comment (VietND96) [reply .idea/misc.xml]: I have reverted it, unexpected changes while opening the project in IntelliJ with configured SDK

**Pull request merged**